src/models.rs

```diff
@@ -313,6 +342,22 @@ pub struct ClassifiedGeneratedTextResult {
     pub input_tokens: Option<Vec<GeneratedToken>>,
 }
 
+#[derive(Default, Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct GenerationWithDetectionResult {
+    /// Generated text
+    #[serde(rename = "generated_text")]
+    pub generated_text: Option<String>,
```

Comment:
Should this really be optional, or do we always expect the LLM to generate something?